### PR TITLE
Add deprecation warnings for v2_normalizeFormMethod

### PR DESCRIPTION
**Pull request by brophdawg11**

### CODE REVIEW

File: packages/remix-react/browser.tsx [hunk 1]
```diff
@@ -151,6 +151,16 @@ export function RemixBrowser(_props: RemixBrowserProps): ReactElement {
...
+    if (!window.__remixContext.future.v2_normalizeFormMethod) {
+      warnOnce(
+        false,
+        "⚠️  DEPRECATED: Please enable the `future.v2_normalizeFormMethod` flag to " +
+          "prepare for the Remix v2 release. Lowercase `useNavigation().formMethod`" +
+          "values are being normalized to uppercase in v2 to align with the `fetch()` " +
+          "behavior.  For more information, see https://remix.run/docs/hooks/use-navigation"
+      );
+    }
+
```
Comment (mcansh) [inline packages/remix-react/browser.tsx:163]: ooh i didn't even think to include deprecations in the browser console

**Review by mcansh — APPROVED** [review]

File: packages/remix-react/browser.tsx (hunk 1 above)
Comment (brophdawg11) [reply packages/remix-react/browser.tsx:163]: yeah I've only done it for remix-react related flags so far.  Not yet convinced it makes sense to put say a flat routes deprecation log in the console

Comment (mcansh) [reply packages/remix-react/browser.tsx:163]: looks like we also let this log in production 🙈 
image

Comment (mcansh) [reply packages/remix-react/browser.tsx:163]: PR to fix #5874

**Pull request merged**